### PR TITLE
Add lib/vimeo.js tests

### LINKED ISSUE #12: Add Testing
Proper tests should be added

### PR DESCRIPTION
**Pull request by quentinms**
Fixes #12 

Notes:
* I haven't added tests for `filestreamer.js` as it doesn't look like it's been used (replaced by TUS I guess?).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+language: node_js
+matrix:
```
Comment (quentinms) [inline .travis.yml]: I copied the matrix from the [tus-js-client](https://github.com/tus/tus-js-client), as it's our only dependency I figured we could use the same. (And maybe add a `latest` too).

File: lib/vimeo.js [hunk 2]
```diff
@@ -326,7 +326,7 @@ Vimeo.prototype.accessToken = function (code, redirectUri, fn) {
  *
  * @param  {string} redirectUri   The URI that will exchange a code for an access token. Must match
  *                                the URI in your API app settings.
- * @param  {string} scope         An array of scopes. See https://developer.vimeo.com/api/authentication#scopes
+ * @param  {string} scope         A space-separated list of scopes. See https://developer.vimeo.com/api/authentication#scopes
```
Comment (quentinms) [inline lib/vimeo.js]: An array works, but the type is `string` so I changed the description. We could also change the type to `string|string[]`.

File: test/lib/vimeo_test.js [hunk 3]
```diff
@@ -0,0 +1,642 @@
+/* eslint-env mocha */
```
Comment (quentinms) [inline test/lib/vimeo_test.js]: This is how `standard` [knows about mocha](https://github.com/standard/standard#what-about-mocha-jasmine-qunit-etc).

File: README.md [hunk 4]
```diff
@@ -205,7 +205,7 @@ Internally, this library executes a `tus` upload approach and sends a file to th
 Name               | Type              | Description
 -------------------|-------------------|------------
 `path`             | string            | Full path to the upload file on the local system.
-`params`           | object (optional) | Parameters to send when creating a new video (name, privacy restrictions, and so on). See the [`/me/videos` documentation](https://developer.vimeo.com/api/endpoints/videos#POST/users/{user_id}/videos) for supported parameters.
+`params`           | object (optional) | Parameters to send when creating a new video (name, privacy restrictions, and so on). See the [`/me/videos` documentation](https://developer.vimeo.com/api/reference/videos#upload_video) for supported parameters.
```
Comment (erunion) [inline README.md]: 👀

File: lib/vimeo.js (hunk 2 above)
Comment (erunion) [reply lib/vimeo.js]: Let's change it to `string|string[]`.

File: test/lib/vimeo_test.js [hunk 5]
```diff
@@ -0,0 +1,642 @@
...
+
+describe('Vimeo.buildAuthorizationEndpoint', () => {
+  const REDIRECT_URL = 'https://myapp.com/login'
+  const vimeo = new Vimeo('id', 'secret', 'token')
+
+  it('uses `public` scope by default', () => {
+    const url = vimeo.buildAuthorizationEndpoint(REDIRECT_URL)
+    expect(url).to.contain('scope=public')
+  })
+  it('uses a space-separated list for scopes', () => {
```
Comment (erunion) [inline test/lib/vimeo_test.js]: Can you add some line breaks between these test cases throughout this file?

**Comment by erunion** [comment]
Correct that `filestreamer.js` is no longer in use since we moved to Tus uploading. I'll remove that after I merge this in.

File: package.json [hunk 6]
```diff
@@ -21,10 +21,16 @@
...
+    "sinon": "^7.x.x",
     "standard": "*"
   },
   "scripts": {
-    "standard": "standard index.js lib/*.js example/*.js",
-    "test": "npm run standard"
+    "standard": "standard index.js lib/*.js example/*.js test/**/*.js",
+    "mocha": "nyc mocha --recursive --reporter landing --timeout=3000",
+    "test": "npm run standard && npm run mocha",
+    "report-coverage": "nyc report --reporter=text-lcov > coverage.lcov && codecov"
```
Comment (erunion) [inline package.json]: Can you add `codecov` as a dev dependency?

```
$ npm run report-coverage

> vimeo@2.1.1 report-coverage /Users/ursenbachj/code/vimeo.js
> nyc report --reporter=text-lcov > coverage.lcov && codecov

sh: codecov: command not found
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! vimeo@2.1.1 report-coverage: `nyc report --reporter=text-lcov > coverage.lcov && codecov`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the vimeo@2.1.1 report-coverage script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/ursenbachj/.npm/_logs/2019-01-18T14_06_45_875Z-debug.log
```

Comment (quentinms) [reply package.json]: Done. Sorry about that.

**Pull request merged**